### PR TITLE
Support :server-ip and :server-hostname as values for :websocket-host

### PR DESCRIPTION
**Pull request by aiba**
Here's my crack at allowing `{:websocket-host :server-ip}` (see #368).

This is a draft for feedback.  If you like it, I will add to this PR updates to `README.md`, `CHANGES.md`, and `FigwheelClientOptions.txt`. 

In the course of implementing this, I found `update-figwheel-connect-options` to be a bit hard to follow, so I endeavored to make it a little easier to think about.  In the process, there was a slight behavior change: now if both `:websocket-host` and `:websocket-url` are supplied, it dissoc'es `:websocket-host` (previously did not).

### CODE REVIEW

**Comment by bhauman** [comment]
This looks great.  Feel free to update the docs.

**Comment by aiba** [comment]
OK, documentation updated.  I didn't edit `CHANGES.md` since I'm not sure which release number heading this would go under.

I tested the one modified function (`update-figwheel-connect-options`) in the REPL and it works on all the test cases I threw at it, but I have to confess I don't know how to test a lein plugin locally, so I haven't done a full end-to-end test.
